src/main/kotlin/org/wycliffeassociates/otter/common/domain/resourcecontainer/project/usfm/UsfmProjectReader.kt

```diff
@@ -33,29 +45,52 @@ class UsfmProjectReader: IProjectReader {
         return Pair(result, projectTree)
     }
 
+    private fun constructTreeFromZip(container: ResourceContainer, project: Project): Pair<ImportResult, Tree> {
+        // Find the appropriate zip entry and use it to construct the project tree
+        ZipFile(container.file).use { zip ->
+            zip.getEntry(project.path.substringAfter("/"))?.let {
```

Comment:
orrrrr... why not [`File(project.path).normalize`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/normalize.html)?

Reply:
I like the File(project.path).normalize approach. The only caveat is that if the path starts with just "/" instead of "./", then it will not remove the "/". For your second option, I don't think we can guarantee that the ZipFile.separator is always going to be "/"?

This would lead me to believe that a regex is the safest option, but I would probably even modify it to make the period optional at the start.

Reply:
What you're proposing would have been fine, but what you did is also fine. I don't think we need to guard against paths starting with `/`, but no problem to do so.